lib/travis/build/script/r.rb

```diff
@@ -0,0 +1,368 @@
+            return
+          end
+          setup_bioc
+          sh.echo "Installing bioc packages: #{packages.join(', ')}"
+          pkg_arg = packages_as_arg(packages)
+          install_script = [
+            "source(\"#{config[:bioc]}\")",
+            'options(repos=biocinstallRepos())',
+            "biocLite(#{pkg_arg})",
+          ].join('; ')
+          sh.cmd "Rscript -e '#{install_script}'"
+        end
+
+        def install_deps
+          setup_devtools
+          if not needs_bioc?
+            install_script = [
+              'library("devtools")',
+              "options(repos = c(CRAN = \"#{config[:cran]}\"))",
+              'install_deps(dependencies = TRUE)',
+            ].join('; ')
+          else
+            install_script = [
+              'library("devtools")',
+              'options(repos = biocinstallRepos())',
+              'install_deps(dependencies = TRUE)',
+            ].join('; ')
+          end
+          sh.cmd "Rscript -e '#{install_script}'"
+        end
+
+        def export_rcheck_dir
+          pkg_script = [
+            'library("devtools")',
+            'cat(paste0(as.package(".")$package, ".Rcheck"))',
+          ].join('; ')
+          sh.export 'RCHECK_DIR', "$(Rscript -e '#{pkg_script}')"
+        end
+        
+        def dump_logs
+          export_rcheck_dir
+          ['out', 'log', 'fail'].each do |ext|
+            cmd = [
+              'for name in',
+              "$(find \"${RCHECK_DIR}\" -type f -name \"*#{ext}\");",
+              'do',
+              'echo ">>> Filename: ${name} <<<";',
+              'cat ${name};',
+              'done',
+            ].join(' ')
+            sh.cmd cmd
+          end
+        end
+        
+        def setup_bioc
+          if not @bioc_installed
+            sh.echo 'Installing BioConductor'
+            bioc_install_script = [
+              "source(\"#{config[:bioc]}\");",
+              'tryCatch(',
+              " useDevel(#{as_r_boolean(config[:bioc_use_devel])}),",
+              ' error=function(e) {if (!grepl("already in use", e$message)) {e}}',
+              ');',
+            ].join(' ')
+            sh.cmd "Rscript -e '#{bioc_install_script}'", retry: true
+          end
+          @bioc_installed = true
+        end
+
+        def setup_devtools
+          if not @devtools_installed
+            devtools_check = '!length(find.package("devtools", quiet = TRUE))'
+            devtools_install = 'install.packages(c("devtools"), ' +
+                               "repos=\"#{config[:cran]}\")"
+            sh.cmd "Rscript -e 'if (#{devtools_check}) #{devtools_install}'"
+          end
+          @devtools_installed = true
+        end
+
+        def setup_latex
+          case config[:os]
+          when 'linux'
+            # We add a backports PPA for more recent TeX packages.
+            sh.cmd 'sudo add-apt-repository -y "ppa:texlive-backports/ppa"'
+
+            sh.cmd 'sudo apt-get install --no-install-recommends ' +
```

Comment:
so that is definitely a good idea -- but how do i make it work with a multi-line list? (i'm guessing that the obvious thing doesn't work based solely on emacs not indenting properly, so let me know if i just need to tell emacs i know better.)